crates/amaru/src/stages/mod.rs

```diff
@@ -51,14 +53,27 @@ pub struct Config {
     pub network: NetworkName,
 }
 
+/// A session with a peer, including the peer itself and a client to communicate with it.
+#[derive(Clone)]
+pub struct PeerSession {
```

Comment:
Is this really the right place for this? Seems to me this is a network-centric concern that perhaps pertains to its own module?

Reply:
Good question. It would maybe make sense to have a separate networking crate for this indeed.